### PR TITLE
[Material] Misc fixes for time picker input mode

### PR DESCRIPTION
**Pull request by rami-a**
## Description

This change:

* Adjusts textfields to ensure they always expand to fill the `SizedBox` parent
* Ensure the hour/minute separator (typically `:`) has its height set to match the hour and minute text fields
* Ensure the hour/minute text fields use the color from the theme when applicable

## Tests

I added the following tests:

* Test to check that the theme is used for the text fields correctly.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/time_picker_theme_test.dart [hunk 1]
```diff
@@ -385,7 +396,7 @@ TimePickerThemeData _timePickerTheme() {
     hourMinuteShape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0))),
     dayPeriodShape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0))),
     dayPeriodBorderSide: const BorderSide(color: Colors.blueAccent),
-    inputDecorationTheme: const InputDecorationTheme(
+    inputDecorationTheme: !includeInputDecoration ? null : const InputDecorationTheme(
```
Comment (clocksmith) [inline packages/flutter/test/material/time_picker_theme_test.dart:399]: I think this is more readable as includeInputDecoration ? cons InputDecorationTheme(...) : null

**Review by clocksmith — APPROVED** [review]

**Review by perclasson — APPROVED** [review]
LGTM

File: packages/flutter/test/material/time_picker_theme_test.dart (hunk 1 above)
Comment (rami-a) [reply packages/flutter/test/material/time_picker_theme_test.dart]: Done

**Review by darrenaustin — APPROVED** [review]
LGTM

**Pull request merged**